### ISSUE TITLE
In libqhull_r, if qh_malloc and qh_free are to be useful, qh_calloc is also needed

### ISSUE THREAD

**Issue by davidheff** [qhull/qhull#121]
usermem_r.c implements qh_malloc and qh_free which are used instead of calls to malloc and free. However, the library also includes three calls to calloc, and there is no corresponding qh_calloc.

In order for qh_malloc and qh_free to be useful, the user might want to replace use of standard malloc and free with functions from another library.  For that to be useful, all memory allocation needs to be user customisable.  So there really needs to be a qh_calloc, and the three calls to calloc to be replaced by calls to qh_calloc also.

**Comment by davidheff** [comment]
Of course, another way is to replace the three calls to calloc with malloc + memset

**Comment by cbbarber** [comment]
Good suggestion.  It will be fixed in the next release (2023.1)

**Comment by davidheff** [comment]
Thanks
